src/react/hooks/useLatestPropsOnEffect.ts

```diff
@@ -0,0 +1,31 @@
+import type { BoardProps } from 'boardgame.io/react';
+import { useEffect, useState } from 'react';
+import { useEffectListener } from './useEffectListener';
+import { useBoardProps } from './useBoardProps';
+
+/**
+ * Returns the latest board props when one or more effect
+ * is triggered. Essentially sugar around `useEffectListener`.
+ * @param effectTypes - List of effects to subscribe to.
+ * @return The boardgame.io props including G and ctx
+ */
+export function useLatestPropsOnEffect(...effectTypes: Array<string>) {
```

Comment:
Here’s a suggestion for improving the typing of the signature.

1. Add an optional type parameter for `G`, so the hook can return `BoardProps<G>` instead of the implicit `BoardProps<any>`

2. Add an optional type parameter for the effects config, which would help type check valid effect names if provided.

```suggestion
import type { BuiltinEffect, EffectsPluginConfig } from '../../types';

type EffectType<C extends EffectsPluginConfig> =
  | BuiltinEffect
  | '*'
  | keyof C['effects'];

/**
 * Returns the latest board props when one or more effect
 * is triggered. Essentially sugar around `useEffectListener`.
 * @param effectTypes - List of effects to subscribe to.
 * @return The boardgame.io props including G and ctx
 */
export function useLatestPropsOnEffect<
  G extends any = any,
  C extends EffectsPluginConfig = EffectsPluginConfig
>(...effectTypes: EffectType<C>[]): BoardProps<G> {
```

Reply:
The looked into doing that for the EffectType but couldn't find something in the code to guide me about allowing `*` and the `effects:` effects but that sorts it nicely. I have to admit, I'm not clear why we need to return the type `BoardProps<G>` instead of `BoardProps` but I'm happy to!

Reply:
Ah, basically the default board props returns `{ G: any, ... }`, so if you have some interface for you game state — like `{ score: number, players: PlayerMap }` or whatever — you don’t get typechecking unless there’s a way to specify what you expect `G` to be. (Of course, it’s only really a guarantee of `G`’s type if you’re consistent about using your custom interface everywhere in your codebase.)